legacy/firmware/fsm.c

```diff
@@ -98,10 +104,11 @@ static uint8_t msg_resp[MSG_OUT_DECODED_SIZE] __attribute__((aligned));
     return;                 \
   }
 
-#define CHECK_UNLOCKED         \
-  if (!session_isUnlocked()) { \
-    layoutHome();              \
-    return;                    \
+#define CHECK_UNLOCKED                                              \
+  if (!session_isUnlocked()) {                                      \
+    fsm_sendFailure(FailureType_Failure_ProcessError, _("Locked")); \
```

Comment:
The test suite doesn't cover it I guess. It's used in non-initial messages like `*Ack`. Normal workflows don't trigger this, because the device should be unlocked by the initial message. I think I accidentally triggered this and without sending the failure the host doesn't realize that something went wrong and waits indefinitely for an answer from the device.